include/deal.II/base/utilities.h

```diff
@@ -50,18 +50,38 @@ namespace Utilities
 {
 
   /**
-   * Convert a number @p i to a string, with as many digits as given to fill
+   * Convert a number @p value to a string, with as many digits as given to fill
    * with leading zeros.
    *
    * If the second parameter is left at its default value, the number is not
-   * padded with leading zeros. The result is then the same as of the standard
+   * padded with leading zeros. The result is then the same as if the standard
    * C function <code>itoa()</code> had been called.
+   *
+   * When calling this function signed integers are implicitly converted to
+   * unsigned integers and long integers might experience an overflow.
+   *
+   * @note The use of this function is discouraged and users should use
+   * <code>Utilities::to_string()</code> instead. In its current implementation
+   * the function simply calls <code>to_string<unsigned int>()</code>.
    */
   std::string
-  int_to_string (const unsigned int i,
+  int_to_string (const unsigned int value,
                  const unsigned int digits = numbers::invalid_unsigned_int);
```

Comment:
There is little work involved in maintaining this one redundant function. I think in principle we all want to keep the interface clean and small, but the reason is typically because maintaining an old interface is cumbersome because it gets in the way of newer things that don't quite fit into an existing interface. An example could be #334 where the existing interface actively gets in the way of changing underlying data structures, and so the only way forward would be to change the interface, deprecate the old one, eventually remove the old one, and then change the data structures.

That doesn't seem to be the case here: nothing is made more complicated or burdensome to maintain by keeping the existing function. On the other hand, we would impose a significant hassle on the many existing users. So my vote would be to simply keep it.